src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -12,741 +24,679 @@ CreateSimilarityIdxCtrl.$inject = ['$scope', 'toastr', '$uibModal', '$timeout',
-                    timer = $timeout($scope.fixSizesOnHorizontalViewModeSwitch, 100);
-                }
-
-                $scope.$on('$destroy', function () {
-                    $timeout.cancel(timer);
-                });
-            }
-        } else {
-            if ($scope.viewMode === 'yasr') {
-                let newHeight = visibleWindowHeight() - (document.querySelector('.CodeMirror-wrap').getBoundingClientRect().top);
-                newHeight -= 40;
-                document.querySelector('.CodeMirror-wrap').style.height = newHeight + 'px';
+        if ($scope.allSamples) {
+            if (SimilarityIndexType.PREDICATION === similarityIndexType) {
+                const allSampleElement = $scope.allSamples[SimilarityIndexType.PREDICATION]['predication'];
+                similarityIndexInfo.setQuery(allSampleElement, SimilarityQueryType.DATA);
             } else {
-                $scope.noPadding = {};
-                document.querySelector('.CodeMirror-wrap').style.height = '';
+                const allSampleElement1 = $scope.allSamples[SimilarityIndexType.TEXT]['literals'];
+                similarityIndexInfo.setQuery(allSampleElement1, SimilarityQueryType.DATA);
             }
-            document.getElementById('yasr').style.minHeight = '';
-        }
-        if (window.yasr && window.yasr.container) {
-            $timeout(function () {
-                window.yasr.container.resize();
-            }, 100);
-        }
-    };
-    $scope.changeViewMode = (tabID) => {
-        $scope.viewMode = 'none';
-        $scope.orientationViewMode = !$scope.orientationViewMode;
-        $scope.fixSizesOnHorizontalViewModeSwitch();
-        $('.dataTables_filter').remove();
-        $('.resultsTable').remove();
-        $timeout(function () {
-            $scope.loadTab();
-            selectTab(tabID);
-        }, 100);
-    };
-    $scope.showHideEditor = () => $scope.fixSizesOnHorizontalViewModeSwitch(true);
-    $scope.focusQueryEditor = () => {
-        if (!angular.element(document).find('.editable-input').is(':focus')) {
-            angular.element(document).find('.CodeMirror textarea:first-child').focus();
         }
-    };
-
-
-    $scope.getResultsDescription = function () {
-    };
-
-    $scope.getUpdateDescription = function () {
-    };
+    }
 
-    $scope.getStaleWarningMessage = function () {
-    };
+    /**
+     * Initializes the similarity index info object with default values if view is opened for create or populates values from url if
+     * view is opened for edit.
+     *
+     * @return {Promise<SimilarityIndexInfo>}
+     */
+    const getSimilarityIndex = () => {
+        const similarityIndexInfo = new SimilarityIndexInfo();
+        similarityIndexInfo.setName(getSimilarityIndexName());
+        similarityIndexInfo.setType(getSimilarityType());
+        similarityIndexInfo.selctedQueryType = $scope.isEditViewMode() ? SimilarityQueryType.SEARCH : SimilarityQueryType.DATA;
```

Comment:
this can also be in the SimilarityIndexInfo model

Reply:
Done